geode-core/src/main/java/org/apache/geode/cache/client/internal/ExecuteRegionFunctionOp.java

```diff
@@ -108,11 +111,19 @@ private static void execute(ExecutablePool pool, String region, String function,
 
         isReexecute = true;
         resultCollector.clearResults();
+        failedNodes = allocateFailedNodesIfEmpty(failedNodes);
         failedNodes.clear();
       }
     } while (true);
   }
 
+  private static Set<String> allocateFailedNodesIfEmpty(Set<String> failedNodes) {
```

Comment:
While it's true that this method allocates a new object sometimes, it seems to me that what it is really doing is ensuring that the set that it returns is mutable. Maybe it would be clearer to make the signature something like `Set<String> ensureMutability(final Set<String> failedNodes)`?

Reply:
@Bill I agree with you. Even if the proposed name is not perfect as you could still pass an immutable non-empty Set and the method will not convert it to a mutable Set, I think it is better than the original and good enough.
Thanks!